packages/mwp-app-render/src/components/uxcapture/UXCaptureConfig.jsx

```diff
@@ -0,0 +1,37 @@
+// @flow
+import React from 'react';
+
+type callbackType = (label: string) => mixed;
+type Props = {|
+	onMark?: callbackType,
+	onMeasure?: callbackType,
+|};
+
+export default ({ onMark, onMeasure }: Props) => {
+	const config = {};
+
+	// No need to inject `UX.config()` script if no callbacks provided
+	if (!onMark && !onMeasure) {
+		return null;
+	}
+
+	if (onMark) {
+		config['onMark'] = onMark.toString();
```

Comment:
@mmcgahan We need to use a combination of `toString()` + `JSON.stringify()` in order to `__dangerouslySetInnerHTML` below. However, the result is script that has the callback as a string instead of a function:

```
<script>
if (window.UX){
    window.UX.config({
        "onMark": "function(mark){ ....}",
    });
}
</script>
```

I'm trying to think of solutions using `eval` (something else?) but coming up short. Any thoughts?

Reply:
I think this issue exposes a potential issue with `onMark` and `onMeasure` in general - since they are part of server-side markup and executed outside of the React application, they only data they can access is the `window` context (including `window.UX`) - they can't, for example, call `setState` or access props on their parent object when the callback executes, which is a _very_ subtle constraint that is going to be difficult to communicate to consumers. However, it also constrains the utility of `onMark` and `onMeasure` to the point that I suspect we don't need to expose them to consumers at all, in which case we probably don't need `UXCaptureConfig.jsx` at all.

To demo what I mean, there's a critical flaw in this code, but it's impossible to see without understanding the implementation details of `UXCaptureConfig`:

```jsx
const Foo = ({ doSomething }) => (
  <UXCaptureConfig onMark={() => doSomething()} />
);
```

This would be completely reasonable code in regular React, but the markup it would generate is something like:

```html
<script>window.UX.config({ onMark: () => doSomething() });</script>
```

when `onMark` was fired, `() => doSomething()` would throw an error because `doSomething` is undefined in the 'raw' HTML/`window` context.

Basically, the _only_ behavior that is allowed in `onMark` and `onMeasure` is behavior that directly interacts with vanilla `window`-based JS APIs, which is something that most frameworks (but particularly React) are designed to avoid.

SO: before getting too deep into implementation tradeoffs, I want to check that exposing 'global' `onMark` and `onMeasure` handlers to the consumer interface through `UX.config` is actually something we want to do. If I remember right, I _think_ that the main purpose of those callbacks is to send the data to New Relic, which I think can still work as long as we can access the `window.newrelic` object, but I think we might want to sort of hard-code that call into our toolset rather than leave it completely open to consumers.